ruma-events/src/room/message/content_serde.rs

```diff
@@ -1,24 +1,56 @@
-//! `Deserialize` implementation for MessageEventContent
+// //! `Deserialize` implementation for MessageEventContent
 
 use serde::{de, Deserialize};
 use serde_json::value::RawValue as RawJsonValue;
 
-use crate::{from_raw_json_value, room::message::MessageEventContent};
+use crate::{
+    from_raw_json_value,
+    room::message::{MessageEventContent, MessageType, Relation},
+};
+
+/// Helper struct to determine the msgtype, relates_to and new_content fields
+/// from a `serde_json::value::RawValue`
+#[derive(Debug, Deserialize)]
+struct MessageContentDeHelper {
+    #[serde(rename = "m.relates_to")]
+    relates_to: Option<Relation>,
+
+    #[cfg(feature = "unstable-pre-spec")]
+    #[serde(rename = "m.new_content")]
+    new_content: Option<MessageEventContent>,
```

Comment:
This needs to be boxed AFAICT (see error in CI).

Reply:
Ahhh there we go I wasn't turning that feature on in the tests. I knew I should have waited to push after class instead of hurrying to try and finish it. Oh well should be good now!